### ISSUE TITLE
Create function dependency map

### ISSUE THREAD

**Issue by jananiravi** [JRaviLab/MolEvolvR#26]
# Create function dependency map for MolEvolvR
- [ ] Identify user-facing functions #7 
- [ ] Identify internal use functions
- [ ] Identify dependency directionality
- [ ] Create function map
- [ ] Automate
- [ ] Resolve #7 | rinse & repeat w/ cleaned up functions

**Comment by SunSummoner** [comment]
@jananiravi I wanted to clarify that I have to tag the functions, find out dependencies and create a function dependency map network. Is there anything more or anything I got wrong?

**Comment by akwe-afriitech** [comment]
Hi @jananiravi , would it be possible for me to be assigned to this task as well?

**Comment by jananiravi** [comment]
Yes, @akwe-afriitech, check in with @SunSummoner's status and you can start contributing to this.

**Comment by akwe-afriitech** [comment]
hey @SunSummoner, please what is your current statues on the tasks, so far i have being able to list all the functions, Identified the user-facing functions
, Identified the internal use functions
 and built a table for dependency directionality, i am currently working on the function map.

**Comment by SunSummoner** [comment]
Hi @akwe-afriitech That sounds great! I have been working on tagging the both the types of functions. Have you done some work on that?

**Comment by Seyi007** [comment]
Hi @SunSummoner @akwe-afriitech, I've made some suggestions with new and more descriptive function names to be updated upon approval. Would this affect your dependency map? And you can find the list of potential new names in the #7 comments. Please let me know if you need any clarifications!

**Comment by akwe-afriitech** [comment]
Hello @SunSummoner, i had gone through all the R files and created a list of all the functions in them, i then wrote a function to convert all function names on the list from snake case and function names case to camel case, of cause it doesn't makes it fully descriptive but it was the quickest way to camel case. Here is what i currently have
![Screenshot 2024-10-04 100215](https://github.com/user-attachments/assets/3b2dd410-d787-4322-8a72-63aa43242d3f)
The data is i json format, beacuse it was the most easily related from the conversion function which i wrote in JS. My R is still coming alone

Hello @Seyi007, yes that would indeed affect the function map, as the names would need to be change afterwards but thats ok, as soon as your suggestions are approve, an hour or 2 of work can fixed the function map

**Comment by Seyi007** [comment]
@akwe-afriitech Take a look at the comments in #7 , I've converted the names to camel cases and I've also replaced all occurrence of the old function names with the new ones in all files (.R and the others) in a forked branch, waiting for approval to create a pr. I suppose you're to work with the old name and new names by treating them as same (old -> new) to map the dependencies on these function.

**Comment by SunSummoner** [comment]
Hey @akwe-afriitech its looks great. If you don't mind to share the data so I can create the graph using R. Otherwise I can make my list and match it with yours and the I can write the R script for the graph.

**Comment by akwe-afriitech** [comment]
sure here is the json file
[function map.json](https://github.com/user-attachments/files/17260894/function.map.json)

**Comment by akwe-afriitech** [comment]
Hello @SunSummoner please do take at this, i isolated all the fucntions with deppendencies and created a visual representation for them. Of cause i will still have to automate the process to work without manually adding the functions. Hoping to get pointers, were i am going wrong, or what to do better before proceeding

![Rplot](https://github.com/user-attachments/assets/8b9116dd-79c3-4de5-a125-c4d84c45c94a)

**Comment by akwe-afriitech** [comment]
![Screenshot 2024-10-04 194601](https://github.com/user-attachments/assets/fdb5fbb8-a5ef-4c09-a8ef-6b8206cdc334)

**Comment by SunSummoner** [comment]
> Hello @SunSummoner please do take at this, i isolated all the fucntions with deppendencies and created a visual representation for them. Of cause i will still have to automate the process to work without manually adding the functions. Hoping to get pointers, were i am going wrong, or what to do better before proceeding

Did you visualise it using R? Or the pictorically?

**Comment by SunSummoner** [comment]
> Hello @SunSummoner please do take at this, i isolated all the fucntions with deppendencies and created a visual representation for them. Of cause i will still have to automate the process to work without manually adding the functions. Hoping to get pointers, were i am going wrong, or what to do better before proceeding

I am planning on working on it automating while coding for the function map.

**Comment by akwe-afriitech** [comment]
> > Hello @SunSummoner please do take at this, i isolated all the fucntions with deppendencies and created a visual representation for them. Of cause i will still have to automate the process to work without manually adding the functions. Hoping to get pointers, were i am going wrong, or what to do better before proceeding
> 
> Did you visualise it using R? Or the pictorically?


i visualised it using R, here is a picture of the code, attaching the code file to a branch of the forked project on my github repos
![Screenshot 2024-10-05 133930](https://github.com/user-attachments/assets/ddaf5e94-c30d-49b1-9787-8b3235e4bc0d)

**Comment by jananiravi** [comment]
@SunSummoner @akwe-afriitech, just to check, are you finding the dependencies manually or automatically, e.g., `pkgdepR`, [`functiondepends`](https://cran.r-project.org/web/packages/functiondepends/functiondepends.pdf), `mvbutils`, `foodweb`, etc.?

**Comment by SunSummoner** [comment]
I was doing it manually but using the packages you suggested is a better alternative.
